utils/s3/aws_errors.hh

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <seastar/core/sstring.hh>
+#include <string>
+
+namespace aws {
+enum class aws_error_type : uint8_t {
+    INCOMPLETE_SIGNATURE = 0,
+    INTERNAL_FAILURE = 1,
+    INVALID_ACTION = 2,
+    INVALID_CLIENT_TOKEN_ID = 3,
+    INVALID_PARAMETER_COMBINATION = 4,
+    INVALID_QUERY_PARAMETER = 5,
+    INVALID_PARAMETER_VALUE = 6,
+    MISSING_ACTION = 7,
+    MISSING_AUTHENTICATION_TOKEN = 8,
+    MISSING_PARAMETER = 9,
+    OPT_IN_REQUIRED = 10,
+    REQUEST_EXPIRED = 11,
+    SERVICE_UNAVAILABLE = 12,
+    THROTTLING = 13,
+    VALIDATION = 14,
+    ACCESS_DENIED = 15,
+    RESOURCE_NOT_FOUND = 16,
+    UNRECOGNIZED_CLIENT = 17,
+    MALFORMED_QUERY_STRING = 18,
+    SLOW_DOWN = 19,
+    REQUEST_TIME_TOO_SKEWED = 20,
+    INVALID_SIGNATURE = 21,
+    SIGNATURE_DOES_NOT_MATCH = 22,
+    INVALID_ACCESS_KEY_ID = 23,
+    REQUEST_TIMEOUT = 24,
+    NOT_INITIALIZED = 25,
+    MEMORY_ALLOCATION = 26,
+
+    NETWORK_CONNECTION = 99,
+    UNKNOWN = 100,
+
+    CLIENT_SIGNING_FAILURE = 101,
+    USER_CANCELLED = 102,
+    ENDPOINT_RESOLUTION_FAILURE = 103,
+    SERVICE_EXTENSION_START_RANGE = 128,
+    OK = -1 // No error set
+};
+
+class aws_error {
+public:
+    aws_error() = default;
+    aws_error(aws_error_type error_type, bool is_retryable);
+    aws_error(aws_error_type error_type, std::string&& error_message, bool is_retryable);
+    const std::string& get_error_message() const;
+    aws_error_type get_error_type() const;
+    bool is_retryable() const;
```

Comment:
probably could just inline the implementation of these accessors?

Reply:
Done